ledger/eval.go

```diff
@@ -1035,6 +1064,38 @@ func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, vali
 	return eval.state.mods, nil
 }
 
+func prefetchThread(ctx context.Context, state roundCowParent, payset []transactions.SignedTxnInBlock) {
+	maybelookup := func(addr basics.Address) {
+		if addr.IsZero() {
+			return
+		}
+		state.lookup(addr)
+	}
+	bail := 10
+	for _, stxn := range payset {
+		state.lookup(stxn.Txn.Sender)
+		maybelookup(stxn.Txn.Receiver)
+		maybelookup(stxn.Txn.CloseRemainderTo)
+		maybelookup(stxn.Txn.AssetSender)
+		maybelookup(stxn.Txn.AssetReceiver)
+		maybelookup(stxn.Txn.AssetCloseTo)
+		maybelookup(stxn.Txn.FreezeAccount)
+		for _, xa := range stxn.Txn.Accounts {
+			maybelookup(xa)
+		}
+		if bail == 0 {
+			bail = 10
+			select {
+			case <-ctx.Done():
```

Comment:
have you observed that calling this `<-ctx.Done()` is making it slow ? It's kind of unexpected that we're doing that only every 10 transactions. ( where the timing isn't really clear )

Reply:
No, just a guess. I probably pulled this pattern from code where just calling time() is enough of a slowdown that it shouldn't be done too often.